src/test/java/com/microsoft/azure/spring/data/cosmosdb/repository/repository/ContactRepository.java

```diff
@@ -10,8 +10,15 @@
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
+import java.util.Optional;
 
 @Repository
 public interface ContactRepository extends CosmosRepository<Contact, String> {
     List<Contact> findByTitle(String title);
+    Iterable<Contact> findByLogicId(String title);
+
+    Contact findOneByTitle(String title);
+
+    Optional<Contact> findOptionallyByTitle(String title);
```

Comment:
Same question as below, what if there are multiple Contacts with same title ?
How would these two APIs behave (`findByName` and this one)..

Reply:
it will fail, as demonstrated by the test `testShouldFailIfMultipleResultsReturned` in the first commit.  it is expected that the client understands their data model and would not add a `findOneBy` or `findOptionallyBy` for queries that could return more than one item.